packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts

```diff
@@ -76,7 +76,14 @@ export class SqsEventSource implements lambda.IEventSource {
     });
     this._eventSourceMappingId = eventSourceMapping.eventSourceMappingId;
 
-    this.queue.grantConsumeMessages(target);
+    // only grant access if the lambda function has an IAM role
+    // otherwise the IAM module will throw an error
+    if (target.role) {
+      this.queue.grantConsumeMessages(target);
+    } else {
+      Annotations.of(target).addWarning(`Function '${target.node.path}' was imported without an IAM role `+
+        `so it was not granted access to consume messages from '${this.queue.node.path}'`);
```

Comment:
If this is already causing an error to be thrown on synth, what is the benefit of adding a warning as well? Aren't these emitted on synth as well?

Reply:
It's the `grantConsumeMessages` that causes an error to be thrown. So this PR essentially changes the error to a warning.